From: Kurt Roeckx <kurt@roeckx.be>
Date: Sat, 18 Jan 2014 11:42:52 +0100
Subject: [PATCH] Always define _GNU_SOURCE

We need this atleast for kfreebsd because they also use glibc.
There shouldn't be a problem defining this on systems not using
glibc.

Imported from openssl_1.0.1f-1.debian.tar.gz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 crypto/dso/dso_dlfcn.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/crypto/dso/dso_dlfcn.c b/crypto/dso/dso_dlfcn.c
index 5f22548..ba97caa 100644
--- a/crypto/dso/dso_dlfcn.c
+++ b/crypto/dso/dso_dlfcn.c
@@ -60,10 +60,8 @@
    that handle _GNU_SOURCE and other similar macros.  Defining it later
    is simply too late, because those headers are protected from re-
    inclusion.  */
-#ifdef __linux
-# ifndef _GNU_SOURCE
-#  define _GNU_SOURCE	/* make sure dladdr is declared */
-# endif
+#ifndef _GNU_SOURCE
+# define _GNU_SOURCE	/* make sure dladdr is declared */
 #endif
 
 #include <stdio.h>
